File: game-server/src/main/java/com/kingston/mmorpg/game/ai/fsm/Run2PatrolTransition.java
package com.kingston.mmorpg.game.ai.fsm;

import com.kingston.mmorpg.game.scene.actor.Creature;

public class Run2PatrolTransition extends Transition {

	public Run2PatrolTransition(State from, State to) {
		super(from, to);
	}

	@Override
	public boolean meetCondition(Creature creature) {
		// 逃跑后肯定是巡逻啦
		return true;
	}

}

File: game-server/src/main/kotlin/org/forfun/mmorpg/game/ai/fsm/Run2PatrolTransition.kt
package org.forfun.mmorpg.game.ai.fsm

import org.forfun.mmmorpg.game.scene.actor.Creature


class Run2PatrolTransition(from: State, to: State) : Transition(from, to) {

    override fun meetCondition(creature: Creature): Boolean {
        // 逃跑后肯定是巡逻啦
        return true
    }
}
